src/scene.js

```diff
@@ -677,9 +681,28 @@ Scene.prototype.forgetTile = function (key) {
 };
 
 
+Scene.prototype.findMaxZoom = function () {
+    var max_zoom = Geo.max_zoom;
+
+    for (var source of this.sources.values()) {
+        if (source.max_zoom > max_zoom) {
+            max_zoom = source.max_zoom;
+        }
+    }
+    return max_zoom;
+};
+
+
+
 // Load a single tile
 Scene.prototype._loadTile = function (coords, options = {}) {
-    var tile = Tile.create({ coords: coords, tile_source: this.tile_source, worker: this.nextWorker() });
+    var tile = Tile.create({
+        coords: coords,
+        max_zoom: this.findMaxZoom(),
```

Comment:
Unsure of the ramifications of this, probably OK for now but I will look into it.